tools/panda.py

```diff
@@ -0,0 +1,140 @@
+    redpanda: RedpandaConfig
+
+
+class Redpanda:
+    def __init__(self, binary, config):
+        self.binary = binary
+        self.config = config
+        self.process = None
+
+    async def run(self):
+        self.process = await asyncio.create_subprocess_exec(
+            self.binary,
+            "--redpanda-cfg",
+            self.config,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT)
+
+        while True:
+            line = await self.process.stdout.readline()
+            if not line:
+                break
+            line = line.decode("utf8").rstrip()
+            print(f"{self.process.pid}: {line}")
+
+        self.process.terminate()
+        return await self.process.wait()
+
+
+async def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-e",
+                        "--executable",
+                        type=pathlib.Path,
+                        help="path to redpanda executable",
+                        default="redpanda")
+    parser.add_argument("--nodes", type=int, help="number of nodes", default=3)
+    parser.add_argument("-d",
+                        "--directory",
+                        type=pathlib.Path,
+                        help="data directory",
+                        default="data")
+    parser.add_argument("--rpc-port", type=int, help="rpc port", default=33145)
+    parser.add_argument("--kafka-port",
+                        type=int,
+                        help="kafka port",
+                        default=9092)
+    parser.add_argument("--admin-port",
+                        type=int,
+                        help="admin port",
+                        default=9644)
+    parser.add_argument("--address",
+                        type=str,
+                        help="listening address",
+                        default="0.0.0.0")
+    args = parser.parse_args()
+
+    seed_servers = [
+        NetworkAddress(args.address, args.rpc_port + i)
+        for i in range(args.nodes)
+    ]
+
+    def make_node_config(i, data_dir):
+        make_address = lambda p: NetworkAddress(args.address, p + i)
+        rpc_address = seed_servers[i]
+        redpanda = RedpandaConfig(data_directory=data_dir,
+                                  rpc_server=rpc_address,
+                                  advertised_rpc_api=rpc_address,
+                                  kafka_api=make_address(args.kafka_port),
+                                  admin=make_address(args.admin_port),
+                                  seed_servers=seed_servers,
+                                  empty_seed_starts_cluster=False)
+        return NodeConfig(redpanda=redpanda)
+
+    def pathlib_path_representer(dumper, path):
+        return dumper.represent_scalar("!Path", str(path))
+
+    def get_config_dumper():
+        d = yaml.SafeDumper
+        d.add_representer(pathlib.PosixPath, pathlib_path_representer)
+        return d
+
+    def prepare_node(i):
```

Comment:
I wonder if the right arch for this is an `rpk` plugin that embeds `gojs` so you still get dynamism and runtime-entry points into the lifecycle of this but the benefit of being able to share w/ wide audience. 

cc: @weeco wdyt of porting that gojs into rpk from console.